### DISCUSSION TITLE
Set timeout using apollo-server

### DISCUSSION THREAD

**Discussion by hugomontero** [apollographql/apollo-server#5736] [Q&A] [answered]
Hi there! 
I was wondering if Is there a reason why setTimeout(...) is not part of apollo-server implementation but is for apollo-server-express?
Is not a good practice to use timeout for our servers or the implementation is kind of different? 

The thing I'm trying to do is change the apollo-server default timeout (All in typescript) and try to send a 504 as response, but I'm not sure if my approach is the correct or maybe I need to change my way to implement apollo-server and stop trying to use specific response status?

Thanks!!! 
P.S. Sorry for my bad communication skills hehe, english is not my mother tongue

**Comment by glasser** [comment, marked as answer]
I'm not sure what timeout you're referring to in `apollo-server-express`.

**Reply by hugomontero** [reply]
Oh! So sorry, What I mean is in Apollo server express I'm able to use something like 
```
const server = app.listen(expressPort, () => {
...
});
server.setTimeout(N)
```

But in Apollo-server  (the standalone) the `server` object in typescript doesn't have the setTimeout function.
And also I didn't see any implementation of change the default 2 minutes timeout in anyplace so my question is something like is there a reason for that? like is not a good practice to do that or do I need to implement the timeout in different way maybe?

**Reply by glasser** [reply]
`apollo-server-express` is the library for when you want to have a lot of control over the detail of how your HTTP server works. `apollo-server` is the library for when you don't need that control and want to be able to type a little bit less to set it up.

**Reply by hugomontero** [reply]
Thanks so much for the explanation @glasser! Now I have more clearness to design my solution. Also thanks for your efforts to keep this libraries up to date!!!
